digitalocean/config/config.go

```diff
@@ -67,11 +73,26 @@ func (c *Config) Client() (*CombinedConfig, error) {
 	})
 
 	userAgent := fmt.Sprintf("Terraform/%s", c.TerraformVersion)
-	client := oauth2.NewClient(oauth2.NoContext, tokenSrc)
+
+	retryableClient := retryablehttp.NewClient()
+	retryableClient.RetryMax = c.HTTPRetryMax
+	retryableClient.RetryWaitMin = time.Duration(c.HTTPRetryWaitMin * float64(time.Second))
+	retryableClient.RetryWaitMax = time.Duration(c.HTTPRetryWaitMax * float64(time.Second))
+
+	client := retryableClient.StandardClient()
+	client.Transport = &oauth2.Transport{
+		Base:   client.Transport,
+		Source: oauth2.ReuseTokenSource(nil, tokenSrc),
+	}
```

Comment:
Thoughts on making this conditional on `c.HTTPRetryMax` being greater than `0`? That way the default behavior remains completely untouched.

Reply:
See no problem with that, I agree it is a safer approach as it would completely avoid any issues that might rise from the retryable client. Updating it now.